packages/compass-collection/src/components/collection-tab-provider.tsx

```diff
@@ -1,9 +1,16 @@
 import React, { useContext, useRef } from 'react';
 import type { CollectionTabPluginMetadata } from '../modules/collection-tab';
+import type { HadronPluginComponent } from 'hadron-app-registry';
 
 export interface CollectionTabPlugin {
   name: string;
-  component: React.ComponentType<CollectionTabPluginMetadata>;
+  component: React.ComponentType<CollectionTabPluginMetadata> &
+    Partial<
+      Pick<
+        HadronPluginComponent<CollectionTabPluginMetadata, any>,
+        'useActivate'
+      >
+    >;
```

Comment:
This looks good but kind of raises the question, could we just use `HadronPluginComponent` as the type of `component` instead of stitching it together from `HadronPluginComponent` and `React.ComponentType` manually?

Reply:
Good callout, thanks, I'm just overthinking this, I was thinking about some cases intel mentioned where potentially they'd want to render their own tabs here that presumably would not need to be hadron plugins. I'll change this to requiring a plugin to be passed to the provider, and we can update later if needed